src/tailscale_svc_lb_controller/resources/base.py

```diff
@@ -0,0 +1,136 @@
+import kubernetes
+
+from src.tailscale_svc_lb_controller import config
+from src.tailscale_svc_lb_controller import helpers
+
+
+class BaseResource:
+    target_service_name = ""
+    target_service_namespace = ""
+    tailscale_proxy_namespace = ""
```

Comment:
Since you have those values here, they are required and all child classes share the same `__init__` method - why not just put the `__init__` here?
```suggestion
    def __init__(self, target_service_name: str, target_service_namespace: str, namespace: str):
        self.target_service_name = target_service_name
        self.target_service_namespace = target_service_namespace
        self.tailscale_proxy_namespace = namespace
```